kinto/core/resource/schema.py

```diff
@@ -150,3 +151,142 @@ class BookmarkSchema(ResourceSchema):
+
+    schema_type = colander.String
+    error_message = "The value should be integer between double quotes"
+    validator = colander.Any(colander.Regex('^"([0-9]+?)"$', msg=error_message),
+                             colander.Regex('\*'))
+
+    def deserialize(self, cstruct=colander.null):
+        param = super(HeaderQuotedInteger, self).deserialize(cstruct)
+        if param is colander.drop or param == '*':
+            return param
+
+        return int(param[1:-1])
+
+
+class HeaderSchema(colander.MappingSchema):
+    """Schema used for validating and deserializing request headers. """
+
+    if_match = HeaderQuotedInteger(name='If-Match', missing=colander.drop)
+    if_none_match = HeaderQuotedInteger(name='If-None-Match', missing=colander.drop)
+    response_behaviour = HeaderField(colander.String(),
+                                     name='Response-Behavior',
+                                     validator=colander.OneOf(['full', 'light', 'diff']),
+                                     missing=colander.drop)
+
+    @staticmethod
+    def schema_type():
+        return colander.Mapping(unknown='preserve')
+
+
+class QuerySchema(colander.MappingSchema):
+    """
+    Schema used for validating and deserializing querystrings. It will include
+    and try to guess the type of unknown fields (filters) on deserialization.
+    """
+
+    _limit = QueryField(colander.Integer())
+    _fields = FieldList()
+    _sort = FieldList()
+    _token = QueryField(colander.String())
+    _since = QueryField(colander.Integer())
+    _to = QueryField(colander.Integer())
+    _before = QueryField(colander.Integer())
+    last_modified = QueryField(colander.Integer())
+
+    @staticmethod
+    def schema_type():
+        return colander.Mapping(unknown='ignore')
+
+    def deserialize(self, cstruct=colander.null):
+        values = {}
+
+        schema_values = super(QuerySchema, self).deserialize(cstruct)
+        if schema_values is colander.drop:
+            return schema_values
+
+        # Deserialize filters
+        for k, v in cstruct.items():
+            # Deserialize lists used on in_ and exclude_ filters
+            if k.startswith('in_') or k.startswith('exclude_'):
+                as_list = FieldList().deserialize(v)
+                if isinstance(as_list, list):
+                    values[k] = [native_value(v) for v in as_list]
+            else:
+                values[k] = native_value(v)
+
+        values.update(schema_values)
+        return values
+
+
+class JsonPatchOperationSchema(colander.MappingSchema):
+    """Single JSON Patch Operation."""
+
+    op = colander.SchemaNode(colander.String(),
+                             validator=colander.OneOf(
+                                 ['test', 'add', 'remove', 'replace', 'move', 'copy']))
+    path = colander.SchemaNode(colander.String())
+    from_ = colander.SchemaNode(colander.String(), name='from', missing=colander.drop)
```

Comment:
That would be cool, but I'm not sure what we can validate here. I know they all have to start with `/`, but I'm not sure if there's anything else to check here. Maybe if there is anything between two `/`? IDK